File: backend/HRManagementSystem/src/Core/HRManagementSystem.Domain/Entities/Notification.cs
using System;
using HRManagementSystem.Domain.Enums.Notification;

namespace HRManagementSystem.Domain.Entities;

public class Notification : BaseEntity
{
    public string? RecipientUserId { get; set; }  // Nullable for system-wide notifications
    public required string Title { get; set; }
    public required string Message { get; set; }
    public NotificationType Type { get; set; }
    public NotificationCategory Category { get; set; }
    public bool IsRead { get; set; } = false;
    public DateTime? ReadAt { get; set; }
    public string? ActionUrl { get; set; }  // Link to related resource
    public int? RelatedEntityId { get; set; }  // e.g., LeaveRequestId, JobApplicationId
    public string? RelatedEntityType { get; set; }  // e.g., "LeaveRequest", "JobApplication"
    public NotificationPriority Priority { get; set; } = NotificationPriority.Normal;

}

File: frontend/src/app/models/notification.ts
export interface Notification {
  id: number;
  recipientUserId?: string;
  title: string;
  message: string;
  type: NotificationType;
  category: NotificationCategory;
  isRead: boolean;
  readAt?: Date;
  actionUrl?: string;
  relatedEntityId?: number;
  relatedEntityType?: string;
  priority: NotificationPriority;
  createdAt: Date;
  updatedAt: Date;
}

export enum NotificationType {
  Info = 'Info',
  Success = 'Success',
  Warning = 'Warning',
  Error = 'Error',
}

export enum NotificationCategory {
  LeaveRequest = 'LeaveRequest',
  LeaveApproval = 'LeaveApproval',
  JobApplication = 'JobApplication',
  EmployeeRegistration = 'EmployeeRegistration',
  Interview = 'Interview',
  Payroll = 'Payroll',
  System = 'System',
}

export enum NotificationPriority {
  Low = 'Low',
  Normal = 'Normal',
  High = 'High',
  Urgent = 'Urgent',
}

export interface CreateNotificationDto {
  recipientUserId?: string;
  title: string;
  message: string;
  type?: NotificationType;
  category: NotificationCategory;
  actionUrl?: string;
  relatedEntityId?: number;
  relatedEntityType?: string;
  priority?: NotificationPriority;
}
